BluetoothLEExplorer/BluetoothLEExplorer/Models/ObservableGattDeviceService.cs

```diff
@@ -46,6 +46,26 @@ public GattDeviceService Service
             }
         }
 
+        /// <summary>
+        /// Lock around the <see cref="Service"/>.
+        /// </summary>
+        SemaphoreSlim serviceLock = new SemaphoreSlim(1, 1);
+
+        public void Dispose()
+        {
+            try

```

Comment:
Don't you only need the lock until you store and clear the member value? 
Use a 'using' / 'lock' statement (if the lock object supports it).
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/lock-statement

Reply:
I can rearrange the code the hold the lock for less time, but I can't use the lock statement because I want to use awaits while creating the object. Additionally, SemaphoreSlim seems to be the right lock out of System.Threading to use, because I don't need cross process communication. However, I don't think that 'using' works with it.